packages/apply/index.ts

```diff
@@ -338,70 +340,20 @@ const bumpMain = ({
-          semver.satisfies(version, errorOnVersionRange)
-        ) {
-          throw new Error(
-            `${pkg.name} will be bumped to ${version}. ` +
-              `This satisfies the range ${errorOnVersionRange} which the configuration disallows. ` +
-              `Please adjust your bump to accommodate the range or otherwise adjust the allowed range in \`errorOnVersionRange\`.`
-          );
-        }
-      }
-    } else if (pkg.vfile.extname === ".toml") {
-      // for rust
-      let version = previewVersion
-      // @ts-ignore
-      ? semver.valid(`${pkg.pkg.package.version}-${previewVersion}`)
-      : semver.inc(
-        // @ts-ignore
-        pkg.pkg.package.version,
-        // @ts-ignore TODO bumpType should be narrowed to meet ReleaseType
-            bumpType,
-            prereleaseIdentifier
-          );
-      if (version) {
-        // @ts-ignore TODO we need to normalize Pkg for toml? Or make some union type
-        pkg.pkg.package.version = version;
-        if (
-          errorOnVersionRange &&
-          semver.satisfies(version, errorOnVersionRange)
-        ) {
-          throw new Error(
-            `${pkg.name} will be bumped to ${version}. ` +
-              `This satisfies the range ${errorOnVersionRange} which the configuration disallows. ` +
-              `Please adjust your bump to accommodate the range or otherwise adjust the allowed range in \`errorOnVersionRange\`.`
-          );
-        }
-      }
-    } else {
-      // assume version is at the root
-      // @ts-ignore TODO bumpType should be narrowed to meet ReleaseType
-      let version = semver.inc(pkg.pkg.version, bumpType);
-      if (version) {
-        pkg.pkg.version = version;
-        if (
-          errorOnVersionRange &&
-          semver.satisfies(version, errorOnVersionRange)
-        ) {
-          throw new Error(
-            `${pkg.name} will be bumped to ${version}. ` +
-              `This satisfies the range ${errorOnVersionRange} which the configuration disallows. ` +
-              `Please adjust your bump to accommodate the range or otherwise adjust the allowed range in \`errorOnVersionRange\`.`
-          );
-        }
-      }
+  const prevVersion = getPackageFileVersion({ pkg });
+  let version =
+    previewVersion && previewVersion !== ""
+      ? semver.valid(`${prevVersion}-${previewVersion}`)
+      : // @ts-ignore TODO bumpType should be narrowed to meet ReleaseType
+        semver.inc(prevVersion, bumpType, prereleaseIdentifier);
+  if (version) {
+    pkg = setPackageFileVersion({ pkg, version });
+    if (errorOnVersionRange && semver.satisfies(version, errorOnVersionRange)) {
+      throw new Error(
+        `${pkg.name} will be bumped to ${version}. ` +
+          `This satisfies the range ${errorOnVersionRange} which the configuration disallows. ` +
+          `Please adjust your bump to accommodate the range or otherwise adjust the allowed range in \`errorOnVersionRange\`.`
+      );
```

Comment:
Do we need that if conditional? Seems version would always be truthy?

Reply:
I am not thrilled with it, but `semver.inc()` can return a `null` if the version isn't valid. Now that we have this abstracted out, I think it will probably be good to put more guardrails around this so all the null-ish / invalid versions / optional types are dealt with in one place.